src/net.h

```diff
@@ -782,6 +791,72 @@ class CNode
     std::atomic_bool fPauseRecv{false};
     std::atomic_bool fPauseSend{false};
 
+    bool IsOutboundOrBlockRelayConn() const {
+        switch(m_conn_type) {
+            case ConnectionType::OUTBOUND:
+            case ConnectionType::BLOCK_RELAY:
+                return true;
+            case ConnectionType::INBOUND:
+            case ConnectionType::MANUAL:
+            case ConnectionType::ADDR_FETCH:
+            case ConnectionType::FEELER:
+                return false;
+            default:
+                assert(false);
+        }
+    }
+
+    bool IsFullOutboundConn() const {
+        return m_conn_type == ConnectionType::OUTBOUND;
+    }
+
+    bool IsManualConn() const {
+        return m_conn_type == ConnectionType::MANUAL;
+    }
+
+    bool IsBlockOnlyConn() const {
+        return m_conn_type == ConnectionType::BLOCK_RELAY;
+    }
+
+    bool IsFeelerConn() const {
+        return m_conn_type == ConnectionType::FEELER;
+    }
+
+    bool IsUniqueNetBlockConn() const {
+        // Netgroups for inbound and manual peers are not excluded because our goal here
+        // is to not use multiple of our limited outbound slots on a single netgroup
+        // but inbound and manual peers do not use our outbound slots.  Inbound peers
+        // also have the added issue that they could be attacker controlled and used
+        // to prevent us from connecting to particular hosts if we used them here.
+        switch(m_conn_type) {
+            case ConnectionType::INBOUND:
+            case ConnectionType::MANUAL:
+                return false;
+            case ConnectionType::OUTBOUND:
+            case ConnectionType::BLOCK_RELAY:
+            case ConnectionType::ADDR_FETCH:
+            case ConnectionType::FEELER:
+                return true;
+            default:
+                assert(false);
+        }
+    }
```

Comment:
I think remove the `default:` (so that we get a warning when a new enum is added and this code not updated to handle it) and put the `assert(false)` after the `switch` (so that the compiler stops complaining about reaching end of non-void function).

If some compiler is not smart enough to see that `assert(false)` is not going to return, then use [abort()](https://en.cppreference.com/w/cpp/utility/program/abort) which is marked as `[[noreturn]]`.

Reply:
updated. I think this will also appease the compiler, but will wait for travis build to confirm..